File: 1886/solution.php
class Solution {

    /**
     * @param Integer[][] $mat
     * @param Integer[][] $target
     * @return Boolean
     */
    function findRotation($mat, $target) {
        $rot = $mat;
        for ($i = 0; $i < 4; $i++) {
            if ($rot == $target) {
                return true;
            }
            $rot = $this->rotate($rot);
        }
        return false;
    }

    private function rotate($mat) {
        // do nothing, placeholder like Python pass
        return $mat;
    }
}

File: 1886/solution.py
def rotate(mat):
    n = len(mat)
    return [
        [mat[-(j + 1)][i] for j in range(n)]
        for i in range(n)
    ]

class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        current = mat

        for _ in range(4):
            current = rotate(current)
            if current == target:
                return True
        return False
